django/db/models/functions/window.py

```diff
@@ -0,0 +1,119 @@
+from django.db.models import FloatField, Func, IntegerField
+
+__all__ = [
+    'CumeDist', 'DenseRank', 'FirstValue', 'Lag', 'LastValue', 'Lead',
+    'NthValue', 'Ntile', 'PercentRank', 'Rank', 'RowNumber',
+]
+
+
+class WindowFunction(Func):
```

Comment:
I think having a separate base class for setting a single flag is probably overkill. If there are future changes that require more customisation then we can introduce a base class to keep things tidy. In the mean time let's not.